pkg/conn/basedb.go

```diff
@@ -64,10 +64,16 @@ func (d *DefaultDBProviderImpl) Apply(config config.DBConfig) (*BaseDB, error) {
 
 	doFuncInClose := func() {}
 	if config.Security != nil && len(config.Security.SSLCA) != 0 {
-		tlsConfig, err := toolutils.ToTLSConfig(config.Security.SSLCA, config.Security.SSLCert, config.Security.SSLKey)
+		tlsConfig, err := toolutils.ToTLSConfigWithVerifyByRawbytes(config.Security.SSLCABytes,
+			config.Security.SSLCertBytes, config.Security.SSLKEYBytes, config.Security.CertAllowedCN)
 		if err != nil {
 			return nil, terror.ErrConnInvalidTLSConfig.Delegate(err)
 		}
+		// NOTE for local test(use a self-signed or invalid certificate), we don't need to check CA file.
+		// see more here https://github.com/go-sql-driver/mysql#tls
+		if config.Host == "127.0.0.1" {
```

Comment:
This check is some what very hack. If we don't want to enable tls in local test, then why set the tls configs? And if the localhost mysql instance is enabled tls, I don't think we should ignore it.